### PR TITLE
Added DMAS and sDMAS 

### PR DESCRIPTION
**Pull request by kdreher**
Main Changes:
- Added Delay-multiply-and-sum (DMAS) as a reconstruction algorithm (Matrone et al.)
- Added the signed DMAS version (Kirchner et al.) as a parameter of DMAS not as its own algorithm since it is only 1 line of code
- Moved the `compute_delay_and_sum_values` function to its own `delay_and_sum` module

### CODE REVIEW

**Review by jgroehl — COMMENTED** [review]
Nice job! Thanks, @kdreher!!
Looks good and is working for me, too!

image

However, there seems to be an issue with the log transform on the DMAS data (not sDMAS, though)..:

image

I'm not sure how this happens, but I think this bug might be independent of this PR?
What do you think?

**Comment by jgroehl** [comment]
lol `envelope` was just set to `True` for DMAS.
Can be merged!

File: image_reconstruction/reconstruction_algorithms/delay_multiply_and_sum.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+                       **kwargs):
+        """
+        Implementation of a delay and sum algorithm.
+
+        The baseline implementation reflects the reconstruction algorithm described by Xu and Wang, 2005::
+
+            Xu, Minghua, and Lihong V. Wang.
+            "Universal back-projection algorithm for photoacoustic computed tomography."
+            Physical Review E 71.1 (2005): 016706.
+
```
[the remark this answers is not in the document]
Comment (kdreher) [reply image_reconstruction/reconstruction_algorithms/delay_multiply_and_sum.py:31]: Oh I might have missed that when I copied the new documentation over from the backprojection file as an example

File: image_reconstruction/reconstruction_algorithms/delay_multiply_and_sum.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+SPDX-FileCopyrightText: 2021 Division of Intelligent Medical Systems, DKFZ
+SPDX-FileCopyrightText: Janek Groehl
+SPDX-License-Identifier: MIT
+"""
+import numpy as np
+from image_reconstruction.reconstruction_algorithms import ReconstructionAlgorithm
+from image_reconstruction.reconstruction_utils.beamforming import delay_multiply_and_sum
+
+
+class DelayMultiplyAndSumAlgorithm(ReconstructionAlgorithm):
```
[the remark this answers is not in the document]
Comment (kdreher) [reply image_reconstruction/reconstruction_algorithms/delay_multiply_and_sum.py:17]: I agree, this would probably be the user-friendliest way :)

**Pull request merged**